src/luascript.cpp

```diff
@@ -12216,15 +12218,24 @@ int LuaScriptInterface::luaItemTypeIsStoreItem(lua_State* L)
 int LuaScriptInterface::luaCombatCreate(lua_State* L)
 {
 	// Combat()
-	pushUserdata<Combat>(L, g_luaEnvironment.createCombatObject(getScriptEnv()->getScriptInterface()));
+	pushSharedPtr(L, g_luaEnvironment.createCombatObject(getScriptEnv()->getScriptInterface()));
 	setMetatable(L, -1, "Combat");
 	return 1;
 }
 
+int LuaScriptInterface::luaCombatDelete(lua_State* L)
+{
+	Combat_ptr combat = getSharedPtr<Combat>(L, 1);
```

Comment:
missing &

Reply:
The only way to convert this into a reference is by making combat_ptr const, but if it becomes const it can't be modified, and therefore can't be deleted.

Reply:
@MillhioreBT remove the const from getSharedPtr so it will return a modifiable shared_ptr reference

Reply:
@yamaken93 I think you have to give me some class on how to violate the const word to make it modifiable

Reply:
> how to violate the const word to make it modifiable

why are you pointing the gun to your own head